From d476ff684614070e619746bcaa2198d8ec7deb85 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Sat, 10 Nov 2018 17:12:24 +0100
Subject: [PATCH 5/5] [SQUSH][DNM] Merge tag 'android-9.0.0_r16' into
 lineage-16.0

commit 929cf3da838935a82e5fdc7278af6e630a2387f9
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Oct 19 22:27:00 2018 +0000

    Version bump to PQ1A.181105.017.A1

    Change-Id: I67c2a1db40cddbe953d3c450dc9520a1ea664d9d

commit 50f56547c8976b4bd0fae0d9d39234ea93460c85
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Oct 5 23:18:17 2018 +0000

    Version bump to PQ1A.181105.017

    Change-Id: I084629194e9d3f995e08e75c156d19730fa8b32f

commit 8c3b43718628e4f5183c6998b35fb011e4b8adae
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Oct 5 00:39:54 2018 +0000

    Version bump to PQ1A.181105.016

    Change-Id: Ib8580ede65f721373b53b6050084cc871a9b7867

commit d10e6651dcd98ac353a7f77d5b021eee4d940110
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Oct 2 18:40:57 2018 +0000

    Version bump to PQ1A.181105.015

    Change-Id: Iaca07acaddc846ed3e52d8be19d5612b417fc43f

commit e9896a5b5f93c7b8abc2ba1557e2aa7b96264b04
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Sep 28 23:14:18 2018 +0000

    Version bump to PQ1A.181105.014

    Change-Id: I90f724e01120b979cebca0e751f2131af864f6ad

commit 310f8241d0e1c6a3e20914205978ecc680c3fbb1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Sep 24 22:16:53 2018 +0000

    Version bump to PQ1A.181105.013

    Change-Id: Icda1903479462e315e1fba086438fa51378238c5

commit b964ecfed1fb481c42fa33fd146f6f42d1fd53b9
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Sep 23 16:47:36 2018 +0000

    Version bump to PQ1A.181105.012

    Change-Id: Ib353386a8da7d1b30086fdded51009e9db7c2fd4

commit fe21655b522405531f13fbbf1d35ddcc114ed712
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Sep 20 19:06:12 2018 +0000

    Version bump to PQ1A.181105.011

    Change-Id: I4764c0ffaf5aa342305810652e4522d2fa89581d

commit 76435aa000d82b33b911106f295c99f956427f61
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Sep 17 22:15:44 2018 +0000

    Version bump to PQ1A.181105.010

    Change-Id: I6b7a3bc1faf60497ed0a56b3df480d9c40efe4cc

commit 6b64578252527ce046aea99eb0c196fca75bd1cc
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Sep 12 17:16:01 2018 +0000

    Version bump to PQ1A.181105.009

    Change-Id: Ib146c931b2a4edee5e0f6b212a0bb7fee4d36683

commit 497b599dcd7b62bb0d3b96567fa26a77d752e890
Merge: 54810589a 17236b90a
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Sep 11 22:38:02 2018 +0000

    Make change and version bump to PQ1A.181105.008

    Change-Id: I0844ddc3490fe7c23c0da9516747db13f98e8819

commit 17236b90a81b95e7337173e5d3826fa86f0dd86a
Author: Elisa Pascual Trevino <elisapascual@google.com>
Date:   Thu Sep 6 14:51:33 2018 -0700

    Update platform security string to 2018-11-05 on pi-dev & master
    Bug:114110585
    (cherry picked from commit dd737c096146a5e3507b1b703c6be62b1db391c0)

    Change-Id: Idaa3e14b642321aea09bfe2dd68dbdb71957a3c3

commit 54810589aae5b839c8128e6ede6ec397ba505077
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Sep 11 18:32:04 2018 +0000

    Version bump to PQ1A.181105.007

    Change-Id: I5d5db4349bf2cdd1a1ee04080ddc2f27c444842f

commit 6e99c58ef482bafb3b2be90163e499997aa2ad3d
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Sep 11 16:35:38 2018 +0000

    Version bump to PQ1A.181105.006

    Change-Id: I78563fc056f1a34d918a66c8deded9678881f98c

commit e582c47011b5dce2aac9388ee457ca33f440690e
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Sep 10 23:34:12 2018 +0000

    Version bump to PQ1A.181105.005

    Change-Id: I3cd26b4e52eb47ce45fb52536abbcdcdb782a8c3

commit d6d42ec47a7dd71e74cd938ea847c4c868f25e99
Merge: 10a9f6840 de7bf957a
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Sep 10 23:21:19 2018 +0000

    Make change and version bump to PQ1A.181101.004

    Change-Id: I46131e1a56dfc409761f0f16c960ce2ee32a5ebe

commit de7bf957a9ac654273dc9a62f6d1c4f615b609c9
Author: Elisa Pascual Trevino <elisapascual@google.com>
Date:   Mon Sep 10 22:57:35 2018 +0000

    Revert "Update platform security string to 2018-11-05 on pi-dev & master"

    This reverts commit dd737c096146a5e3507b1b703c6be62b1db391c0.

    Reason for revert: Security String needs to be temporary reverted to cut a new 11-01 build

    Change-Id: I36e0f4de0c4158711fcc7a1f383129b8f3f889a1
    (cherry picked from commit ad151b50f3b51022896ae674970a203b3f4c681e)

commit 10a9f6840a6a209fe9f8fdf5833a5ea179e33268
Merge: 314f8a5d1 5d1f2a53e
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Sep 10 22:35:10 2018 +0000

    Make change and version bump to PQ1A.181105.003

    Change-Id: Iabdd32a8601fb029bee41d3c4e7320d3a600d05e

commit 5d1f2a53ef9bc6bee3f31d85066f70c321046674
Author: Elisa Pascual Trevino <elisapascual@google.com>
Date:   Thu Sep 6 14:51:33 2018 -0700

    Update platform security string to 2018-11-05 on pi-dev & master
    Bug:114110585
    (cherry picked from commit dd737c096146a5e3507b1b703c6be62b1db391c0)

    Change-Id: I0e8d07c6d7333414c6099f718a122d7b343d8e92

commit 314f8a5d1663f7c328902768cd419d4aae490aca
Merge: 95212e441 3e587b55b
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Sep 10 21:36:42 2018 +0000

    Make change and version bump to PQ1A.181101.001

    Change-Id: I53c56b24f920207fa9373e7cc6db67bdb6912307

commit 3e587b55b929e9da5b1892aa931258e2212d9a32
Author: Elisa Pascual Trevino <elisapascual@google.com>
Date:   Thu Sep 6 14:47:53 2018 -0700

    Update platform security string to 2018-11-01 on pi-dev & master
    Bug:114110585
    (cherry picked from commit 2e6c06ba1b01fd0b9af8d955ae28da6b39037c85)

    Change-Id: Ic1e2d648daa0c28d7d203e44fd6bbe0f705e4eab

commit 95212e441073530fb240c30354d4b2b2bb420416
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Sep 7 01:09:33 2018 +0000

    Version bump to PQ1A.180829.006

    Change-Id: I6f759740d8a7759f3437efcbc3837271961a1463

commit cb30f9d83aecae87ee43cc05194b2bef748a14a4
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Sep 5 23:36:14 2018 +0000

    Version bump to PQ1A.180829.005

    Change-Id: I1db4c1e8b98a1a59a8fcbc11a5806e1f34a62b9b

commit 8d80a97de4a37f040113ad0f54bab7c08c633e12
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Aug 31 04:27:29 2018 +0000

    Version bump to PQ1A.180829.004

    Change-Id: I5b2e6f0ef333e7220b9243f4382cc2e925893a44

commit 29a403c99786924564eace8a05df145b219e7ca1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Aug 30 04:03:06 2018 +0000

    Version bump to PQ1A.180829.003

    Change-Id: I5db12a96587363a6c6cef92c0105f5d9c75ea4a0

commit d77732f6218595e7bea4f2da4e1c89c98d55fb9e
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Aug 29 21:55:12 2018 +0000

    Version bump to PQ1A.180829.002

    Change-Id: I52741f1e002b7bb3712d0256c5f8482d97f03ec9

commit b9dbc5248d5fe491be731c10848d95f6fc77949f
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Aug 28 19:21:40 2018 +0000

    Version bump to PQ1A.180824.006

    Change-Id: I68b4ac3017d5ca1c2274075102c69d50c1bf051a

commit 945442b156855a3d40b1bff46b1d355a75e6be30
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Aug 28 04:22:20 2018 +0000

    Version bump to PQ1A.180824.005

    Change-Id: Ia6aa71d667560446c8c668a893a87fab2363814a

commit 876e565e7d9cdde31c5273c62f6cdfab6e33733f
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Aug 28 01:57:43 2018 +0000

    Version bump to PQ1A.180824.004

    Change-Id: I70f5cc40c09d094682e38a29c077edceab9dcb52

commit 45a70dd5a4f470633f0b983cec602a2b1c2167ab
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Aug 24 22:57:41 2018 +0000

    Version bump to PQ1A.180824.003

    Change-Id: I6e56000dcfd0dac22705d8b27b2d12508786efba

commit f82a328141759d2ef672577934436acc8173f953
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Aug 24 22:01:22 2018 +0000

    Version bump to PQ1A.180824.002

    Change-Id: I1e3748d29da864f90f11ccee9e28fb4e7840c571

commit 5a51ff6b6c8fa9ceb19434117c3e399851edd86b
Merge: 75df03bf2 a113745d1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Aug 24 03:06:29 2018 +0000

    Make change and version bump to PQ1A.180824.001

    Change-Id: I442bec7b7a7871f7cbf2438e6ccbf9385ad8b9ba

commit 75df03bf29ec955cd92c5f605d421ae250bd86c0
Merge: b8ad2c0a9 a113745d1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Aug 23 03:03:15 2018 +0000

    Make change and version bump to PQ1A.180823.001

    Change-Id: Ia6cb777aa2f5cb2bed10f09cfde881e85b53f498

commit b8ad2c0a9bf7a6e0705877ba70670d1007a8c0c0
Merge: 6b2be4559 a113745d1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Aug 22 03:14:38 2018 +0000

    Make change and version bump to PQ1A.180822.001

    Change-Id: Ia954dc4b035f31871e11fc68611714b33331fcd2

commit 6b2be4559205542f0b82c1501fac459f53047e1c
Merge: faee5419b a113745d1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Aug 21 03:07:06 2018 +0000

    Make change and version bump to PQ1A.180821.001

    Change-Id: I615c99cc23fe0727cbe070d4b7848dc80bd35962

commit faee5419b10d8c966f630640e53d204d47a3433c
Merge: 5838ce347 a113745d1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Aug 20 03:08:54 2018 +0000

    Make change and version bump to PQ1A.180820.001

    Change-Id: I25d237f221b5980de4806202f60f1574bdaa9109

commit 5838ce3474814c0a1903bf0cf68454b8242150ab
Merge: 54abdc0b6 d8186f43a
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Aug 17 03:00:40 2018 +0000

    Make change and version bump to PQ1A.180817.001

    Change-Id: I600aa2f1a43ea4c8a25c6e89df56d010fa043ee6

commit a113745d1f15a358f427e9997490099e34807377
Author: Colin Cross <ccross@android.com>
Date:   Tue Aug 14 13:13:47 2018 -0700

    Track more static dependencies for notice files

    Track transitive static dependencies of native binaries, and
    and direct and transitive static dependencies of java and app
    modules.

    Bug: 112331930
    Test: m checkbuild
    Change-Id: I0e19971033d6254bfbb6555f0e68fd5e529569c6
    Merged-In: I0e19971033d6254bfbb6555f0e68fd5e529569c6
    (cherry picked from commit 579668b122d88b9c810238412ab003192bf15d2b)

commit 54abdc0b64f4accf11dd38568792375d39244036
Merge: 8aba4494d d8186f43a
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Aug 16 03:07:12 2018 +0000

    Make change and version bump to PQ1A.180816.001

    Change-Id: Ied6b6a68a5af75e83740a41ec7505f3babedba35

commit d8186f43a723c8600e13d6ab9d58014de4f484af
Merge: 61ddf1fa2 090027fb4
Author: Adam Seaton <aseaton@google.com>
Date:   Wed Aug 15 18:32:26 2018 -0700

    [automerger skipped] DO NOT MERGE Update platform security string to 2018-10-05 bug:112535135
    am: 090027fb46  -s ours

    Change-Id: I6d5364d1d0bcbc2960ea259a25ceeab4b24b4ead

commit 8aba4494d5ba080612cfa478981a8b2ec8656462
Merge: 88a5257a3 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Aug 15 03:14:18 2018 +0000

    Make change and version bump to PQ1A.180815.001

    Change-Id: I7abbc908643d6fb7c66111b49dd7ad8a23e57ba7

commit 88a5257a3825f78d0555b1a9e5fd30d4a1ccfbab
Merge: 245a2ad1a 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Aug 14 03:03:59 2018 +0000

    Make change and version bump to PQ1A.180814.001

    Change-Id: Ic3d177140cce85ef04e57b5375527f469f13d491

commit 090027fb461c2a8ed1d3277da4f01e11565865ed
Author: Adam Seaton <aseaton@google.com>
Date:   Mon Aug 13 11:12:19 2018 -0700

    DO NOT MERGE Update platform security string to 2018-10-05
    bug:112535135

commit 61ddf1fa27039bc83a7e4be48a35a50ab2b83c78
Author: Adam Seaton <aseaton@google.com>
Date:   Mon Aug 13 09:34:24 2018 -0700

    Update platform security string to 2018-10-05 on pi-dev & master
    bug: 112535135

commit 245a2ad1ac80d7ddc08e9afc1f4bdf4e5d128218
Merge: 2b106427b 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Aug 13 03:11:06 2018 +0000

    Make change and version bump to PQ1A.180813.001

    Change-Id: If70ddfffd95d5d194dfa961214de43abe96c2861

commit 2b106427b7adac09ddea1606d5ce34ed27b64f57
Merge: 2753ce181 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Aug 10 03:18:35 2018 +0000

    Make change and version bump to PQ1A.180810.001

    Change-Id: I1824ba20426e3fab9b8f6c0829501e55bf6f03ca

commit 2753ce181dc29d7e6ce084379284341bdb2e80c4
Merge: f455fac5c 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Aug 9 03:09:02 2018 +0000

    Make change and version bump to PQ1A.180809.001

    Change-Id: Iaf437cd68abbad424802b4ac2a2b5990c1b51c97

commit f455fac5c4303f685efd971f61485ec8bb63c0f8
Merge: 2354f7c53 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Aug 8 03:11:53 2018 +0000

    Make change and version bump to PQ1A.180808.001

    Change-Id: I1f4115843c57c1c955c807f08c65c906128e4060

commit 2354f7c53eee153d369ef18702d127695f99017b
Merge: dba5b7622 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Aug 7 03:07:58 2018 +0000

    Make change and version bump to PQ1A.180807.001

    Change-Id: If7b9224c67e48f72a658c46d1d503625d876f2c9

commit dba5b76223c1d0f0254c73c77dae34d0545d7e1e
Merge: 9aa201067 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Aug 6 03:02:17 2018 +0000

    Make change and version bump to PQ1A.180806.001

    Change-Id: I8e0af317560ebd1e2f7eb0bec7c9cc6b7277326b

commit 9aa201067987e2155c6433de533a086732d539d0
Merge: 05bef30b1 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Aug 3 03:11:46 2018 +0000

    Make change and version bump to PQ1A.180803.001

    Change-Id: Id87a91cada5b33fdb30295b867e1c8a0dc0f691b

commit 05bef30b126c1b5366496961bfae27bb2ce01934
Merge: bbed13f77 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Aug 2 03:03:37 2018 +0000

    Make change and version bump to PQ1A.180802.001

    Change-Id: Idf01581ca56743664287be7bf9a41303472ee3ca

commit bbed13f77dc21b39f8290b21ab65bc86981dcbf9
Merge: cfe94cdd5 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Aug 1 03:07:49 2018 +0000

    Make change and version bump to PQ1A.180801.001

    Change-Id: I6836f12d465ed28353a47c3015506a31fb40fc0d

commit cfe94cdd56b6b5e3a9cbe17ddff0bd2c76d03f44
Merge: 631c7c688 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Jul 31 03:02:57 2018 +0000

    Make change and version bump to PQ1A.180731.001

    Change-Id: If5b30bde8030a878ca68b59fc4261bf04709ff9c

commit 631c7c6884b2bf385c73f62d680ede91d78e03a9
Merge: adff35964 8c3565d98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Jul 30 03:08:30 2018 +0000

    Make change and version bump to PQ1A.180730.001

    Change-Id: I5c69e56dafb10bf5ff4f6fa70bf1c050bea5b5cd

commit adff35964055d44af08a037b7cc0b47839784be0
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Jul 27 03:15:58 2018 +0000

    Version bump to PQ1A.180727.001

    Change-Id: I917e56aed0af0270449bec17471f74abdc569240

commit 8c3565d98de92490d4e3a7069941458d0e94535a
Author: Mac Wang <macw@google.com>
Date:   Thu Jul 5 15:46:08 2018 +0800

    Add product_launched_with_k.mk

    Bug: 25564210
    Test: None
    Change-Id: I2afabf6d7720e9f1026f86598c4f6bc392a1c888

commit 57c2d9f670af89bdaeb347cd59c9d465f8d29e58
Merge: a83fc0e48 6bc223c9a
Author: Adam Seaton <aseaton@google.com>
Date:   Mon Jul 16 14:22:57 2018 -0700

    Updating Platform Security String to 2018-09-05 Bug: 111501777
    am: 6bc223c9af

    Change-Id: I588810a6254c3c14d9e4398a3be05987308e12af

commit 6bc223c9af044ad06e2f1abc0c4570a7371f9a3a
Author: Adam Seaton <aseaton@google.com>
Date:   Mon Jul 16 10:17:21 2018 -0700

    Updating Platform Security String to 2018-09-05
    Bug: 111501777

commit a83fc0e482a91f7c860198d06dda0d0fac95b28e
Author: bohu <bohu@google.com>
Date:   Mon Jun 25 15:33:14 2018 -0700

    emulator: fix a typo when copying userdata.img

    replaced TARGET_CPU_ADI with TARGET_CPU_ABI

    BUG: 110492064

    this cl does not impact real devices

    Test:
    nohup make -j100 ANDROID_USE_BUILDCACHE=false PRODUCT-sdk_gphone_x86-sdk_addon showcommands dist DIST_DIR=${PWD}/dist/x86 sdk_repo
    unzip the image package sdk-repo-linux-system-images-eng.zip and  make sure userdata.img is inside

    Change-Id: I24101c78d789c9b72132efc41a2e98d5274b5ff1

commit 2276fb95cff97375ca38b6901480a94f8c8806cc
Author: Weilun Du <wdu@google.com>
Date:   Fri Jun 22 13:48:21 2018 -0700

    Copy dummy userdata.img to sdk_addon images.

    Avd manager in Android Studio requires userdata.img in the downloaded
    system image. As we no longer inlcude the real userdata.img, we copy the
    dummy userdata.img to pass the check.
    This cl does not impact real devices.

    BUG: 110492064
    Test: Manual
    1, Build with command "make -j110 PRODUCT-sdk_gphone_x86-sdk_addon".
    2, Check if the dummy userdata.img appears in sdk addon image zip.
    Change-Id: Icb88aac42921502288efe7355b8cd76704d5eb30
    Signed-off-by: Weilun Du <wdu@google.com>

commit bd141d36a0d19971ad44a5df3900cf68422fdb75
Merge: f53de9371 31b2f7e57
Author: Weilun Du <wdu@google.com>
Date:   Mon Jun 18 23:11:40 2018 +0000

    Merge "Increase system image size to 2.5 GB" into pi-dev

commit f53de93711f428033464af9f07405211aa61ae87
Merge: 16e7004e3 f6bee62b8
Author: Adam Seaton <aseaton@google.com>
Date:   Mon Jun 18 12:52:43 2018 -0700

    [automerger skipped] [DO NOT MERGE] Update Platform Security String to 2018-08-05 for August Bug:109889880
    am: f6bee62b8e  -s ours

    Change-Id: Ib4f9aecf86cbf659a9a27d07fd03c7a799518e0e

commit f6bee62b8eb5819f23e31d1dfb1622d48f5ab554
Author: Adam Seaton <aseaton@google.com>
Date:   Mon Jun 18 09:38:40 2018 -0700

    [DO NOT MERGE] Update Platform Security String to 2018-08-05 for August
    Bug:109889880

commit 31b2f7e57f852d96ff94c911f72f4f5926989208
Author: Weilun Du <wdu@google.com>
Date:   Sun Jun 17 15:37:53 2018 -0700

    Increase system image size to 2.5 GB

    Fixed build breakge caused by insufficient image size.
    BUG: 110198136

    Change-Id: I12a4a43b522b7867ff789854cf5b76d13f2f5499
    Signed-off-by: Weilun Du <wdu@google.com>

commit 16e7004e3fc05ecb829e74e19bee4c1bf66c5316
Author: Hyunyoung Song <hyunyoungs@google.com>
Date:   Thu Jun 14 16:08:47 2018 -0700

    Fix unbundled branch by not having it pick up goldfish dependency
    Bug: 110165361

    Change-Id: I0824b140fdfee296288c03fd03a3bac7966e9179

commit 80ca83c41f4f2def5a9e01513a7cd8759c6a8438
Merge: e64b309d2 fbde08eb0
Author: TreeHugger Robot <treehugger-gerrit@google.com>
Date:   Thu Jun 14 01:49:08 2018 +0000

    Merge "emulator: base sdk_phone_armv7/64 on aosp" into pi-dev

commit e64b309d22ba5b40f5e591890b5291e33b2805ff
Merge: d7dd3e2d3 0d749a7a3
Author: TreeHugger Robot <treehugger-gerrit@google.com>
Date:   Wed Jun 13 22:19:14 2018 +0000

    Merge "emulator: move sepolicy to goldfish project" into pi-dev

commit d7dd3e2d3d3d3828b53a688f8cc965df1ba1b60c
Author: bohu <bohu@google.com>
Date:   Wed Jun 13 09:04:30 2018 -0700

    emualtor: cleanup media configuration

    The media configuration is handled by goldfish/vendor.mk
    already; clean them up from device.mk files.

    this cl does not impact arm devices;
    and it does the right thing for x86/64 devices
    because the media configuration should go to vendor
    instead of staying on system

    BUG: 110030159
    Change-Id: If492f1e01f600b1c969e9ef04598f7af6783970e

commit 0d749a7a3ea922bda49f00635e5abbb06b02c558
Author: bohu <bohu@google.com>
Date:   Wed Jun 13 10:18:07 2018 -0700

    emulator: move sepolicy to goldfish project

    The sepolicies are emulator specific and are installed
    under vendor partition, move them to the right location.

    this cl does not impact real devices, as the selinux
    rules are for emulator only

    BUG: 110030159
    Change-Id: I6acc27a3b787a3fafd9373c84492537185b184c5
    Merged-In: I6acc27a3b787a3fafd9373c84492537185b184c5

commit 941f8e102c024f884617baaa9471a27c55bdccb0
Merge: 667685301 a2d754ce3
Author: Bo Hu <bohu@google.com>
Date:   Wed Jun 13 05:18:47 2018 +0000

    Merge "Support new hostapd build target without HIDL" into pi-dev

commit fbde08eb073b690467aadb722da47bc287649dc4
Author: bohu <bohu@google.com>
Date:   Mon Jun 11 23:23:10 2018 -0700

    emulator: base sdk_phone_armv7/64 on aosp

    Switch to GSI images for arm/arm64 sdk images.

    BUG: 79941736

    This cl does not impact real device images

    Test: make -j110 PRODUCT-sdk_phone_armv7-sdk showcommands dist \
    DIST_DIR=git_master-without-vendor-linux-sdk_x86-sdk sdk_repo

    Change-Id: Ia50be068cc5e5307cdf7ee3a8e11014ed8019992

commit 667685301600040c0006752bbf7ec0335efa9a56
Author: bohu <bohu@google.com>
Date:   Sun Jun 10 00:11:30 2018 -0700

    emulator: factor vendor partition out of emulator.mk

    The emulator vendor paritions are better moved out of
    emulator.mk into device/generic/goldfish/vendor.mk
    so that changes in vendor.mk will just concentrate on
    vendor partition.

    this cl does not impact real devices, and it does not
    even impact emulator images (it does code refactoring only)

    BUG: 110030159

    Change-Id: Ic5778cda8ca6ccf7bc2bc2f9028f801cda970143

commit a2d754ce3a611bcd51945e17642cb52e920cc0ad
Author: Bjoern Johansson <bjoernj@google.com>
Date:   Tue May 15 15:03:12 2018 -0700

    Support new hostapd build target without HIDL

    The new build target for hostapd does not include the HIDL control
    interface and it's a separate binary from the hostapd used for WiFi
    hotspots. This new binary needs its own SELinux rules and it should be
    included in emulator builds since it's used to provide virtual access
    points.

    BUG: 74401469
    Test: run cts -m CtsNetTestCases

    (cherry picked from commit a53c522678f9178038dfdda6260f5c1b69c416ba)

    Change-Id: I8fa4908d1bef8ff25573abb72bbac433ae3151de

Change-Id: I7dcd35f6afc0ced9efe8e29167fe0f4f593b624c
---
 core/binary.mk                                |  8 ++--
 core/build_id.mk                              |  2 +-
 core/java_common.mk                           | 15 ++++++
 core/tasks/sdk-addon.mk                       |  1 +
 core/version_defaults.mk                      |  2 +-
 target/board/generic/device.mk                | 11 -----
 target/board/generic/sepolicy/OWNERS          |  4 --
 target/board/generic/sepolicy/adbd.te         |  1 -
 target/board/generic/sepolicy/audioserver.te  |  1 -
 target/board/generic/sepolicy/bootanim.te     |  9 ----
 target/board/generic/sepolicy/cameraserver.te |  2 -
 target/board/generic/sepolicy/createns.te     | 14 ------
 target/board/generic/sepolicy/device.te       |  1 -
 target/board/generic/sepolicy/dhcpclient.te   | 20 --------
 target/board/generic/sepolicy/dhcpserver.te   | 12 -----
 target/board/generic/sepolicy/domain.te       |  3 --
 target/board/generic/sepolicy/execns.te       | 34 --------------
 target/board/generic/sepolicy/file.te         |  4 --
 target/board/generic/sepolicy/file_contexts   | 46 ------------------
 target/board/generic/sepolicy/genfs_contexts  | 20 --------
 .../board/generic/sepolicy/goldfish_setup.te  | 47 -------------------
 .../generic/sepolicy/hal_camera_default.te    |  3 --
 .../board/generic/sepolicy/hal_cas_default.te |  1 -
 .../board/generic/sepolicy/hal_drm_default.te |  2 -
 .../generic/sepolicy/hal_drm_widevine.te      | 14 ------
 .../sepolicy/hal_fingerprint_default.te       |  5 --
 .../generic/sepolicy/hal_gnss_default.te      |  3 --
 .../hal_graphics_allocator_default.te         |  2 -
 .../sepolicy/hal_graphics_composer_default.te |  3 --
 .../generic/sepolicy/hal_wifi_default.te      |  1 -
 target/board/generic/sepolicy/healthd.te      |  2 -
 target/board/generic/sepolicy/init.te         |  2 -
 target/board/generic/sepolicy/ipv6proxy.te    | 16 -------
 target/board/generic/sepolicy/logpersist.te   | 13 -----
 target/board/generic/sepolicy/mediacodec.te   |  1 -
 target/board/generic/sepolicy/netd.te         |  3 --
 target/board/generic/sepolicy/priv_app.te     |  5 --
 target/board/generic/sepolicy/property.te     |  5 --
 .../board/generic/sepolicy/property_contexts  |  8 ----
 target/board/generic/sepolicy/qemu_props.te   | 10 ----
 target/board/generic/sepolicy/radio.te        |  3 --
 target/board/generic/sepolicy/rild.te         |  3 --
 target/board/generic/sepolicy/shell.te        |  1 -
 .../board/generic/sepolicy/surfaceflinger.te  |  5 --
 .../board/generic/sepolicy/system_server.te   |  1 -
 target/board/generic/sepolicy/vendor_init.te  |  1 -
 target/board/generic/sepolicy/vold.te         |  1 -
 target/board/generic/sepolicy/zygote.te       |  5 --
 target/board/generic_arm64/device.mk          | 10 ----
 target/board/generic_x86/device.mk            | 14 ------
 target/board/generic_x86/sepolicy/OWNERS      |  4 --
 target/board/generic_x86/sepolicy/domain.te   |  1 -
 target/board/generic_x86/sepolicy/healthd.te  |  1 -
 target/board/generic_x86/sepolicy/init.te     |  1 -
 target/board/generic_x86/sepolicy/installd.te |  1 -
 target/board/generic_x86/sepolicy/zygote.te   |  2 -
 target/board/generic_x86_64/device.mk         | 14 ------
 target/board/generic_x86_arm/BoardConfig.mk   |  2 +-
 target/product/product_launched_with_k.mk     |  2 +
 target/product/product_launched_with_l.mk     |  1 -
 target/product/sdk_phone_arm64.mk             | 28 ++---------
 target/product/sdk_phone_armv7.mk             | 20 ++------
 62 files changed, 32 insertions(+), 445 deletions(-)
 delete mode 100644 target/board/generic/sepolicy/OWNERS
 delete mode 100644 target/board/generic/sepolicy/adbd.te
 delete mode 100644 target/board/generic/sepolicy/audioserver.te
 delete mode 100644 target/board/generic/sepolicy/bootanim.te
 delete mode 100644 target/board/generic/sepolicy/cameraserver.te
 delete mode 100644 target/board/generic/sepolicy/createns.te
 delete mode 100644 target/board/generic/sepolicy/device.te
 delete mode 100644 target/board/generic/sepolicy/dhcpclient.te
 delete mode 100644 target/board/generic/sepolicy/dhcpserver.te
 delete mode 100644 target/board/generic/sepolicy/domain.te
 delete mode 100644 target/board/generic/sepolicy/execns.te
 delete mode 100644 target/board/generic/sepolicy/file.te
 delete mode 100644 target/board/generic/sepolicy/file_contexts
 delete mode 100644 target/board/generic/sepolicy/genfs_contexts
 delete mode 100644 target/board/generic/sepolicy/goldfish_setup.te
 delete mode 100644 target/board/generic/sepolicy/hal_camera_default.te
 delete mode 100644 target/board/generic/sepolicy/hal_cas_default.te
 delete mode 100644 target/board/generic/sepolicy/hal_drm_default.te
 delete mode 100644 target/board/generic/sepolicy/hal_drm_widevine.te
 delete mode 100644 target/board/generic/sepolicy/hal_fingerprint_default.te
 delete mode 100644 target/board/generic/sepolicy/hal_gnss_default.te
 delete mode 100644 target/board/generic/sepolicy/hal_graphics_allocator_default.te
 delete mode 100644 target/board/generic/sepolicy/hal_graphics_composer_default.te
 delete mode 100644 target/board/generic/sepolicy/hal_wifi_default.te
 delete mode 100644 target/board/generic/sepolicy/healthd.te
 delete mode 100644 target/board/generic/sepolicy/init.te
 delete mode 100644 target/board/generic/sepolicy/ipv6proxy.te
 delete mode 100644 target/board/generic/sepolicy/logpersist.te
 delete mode 100644 target/board/generic/sepolicy/mediacodec.te
 delete mode 100644 target/board/generic/sepolicy/netd.te
 delete mode 100644 target/board/generic/sepolicy/priv_app.te
 delete mode 100644 target/board/generic/sepolicy/property.te
 delete mode 100644 target/board/generic/sepolicy/property_contexts
 delete mode 100644 target/board/generic/sepolicy/qemu_props.te
 delete mode 100644 target/board/generic/sepolicy/radio.te
 delete mode 100644 target/board/generic/sepolicy/rild.te
 delete mode 100644 target/board/generic/sepolicy/shell.te
 delete mode 100644 target/board/generic/sepolicy/surfaceflinger.te
 delete mode 100644 target/board/generic/sepolicy/system_server.te
 delete mode 100644 target/board/generic/sepolicy/vendor_init.te
 delete mode 100644 target/board/generic/sepolicy/vold.te
 delete mode 100644 target/board/generic/sepolicy/zygote.te
 delete mode 100644 target/board/generic_x86/sepolicy/OWNERS
 delete mode 100644 target/board/generic_x86/sepolicy/domain.te
 delete mode 100644 target/board/generic_x86/sepolicy/healthd.te
 delete mode 100644 target/board/generic_x86/sepolicy/init.te
 delete mode 100644 target/board/generic_x86/sepolicy/installd.te
 delete mode 100644 target/board/generic_x86/sepolicy/zygote.te
 create mode 100644 target/product/product_launched_with_k.mk

diff --git a/core/binary.mk b/core/binary.mk
index 5c26e5f72..5962f9d2c 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -1624,6 +1624,9 @@ installed_static_library_notice_file_targets := \
     $(foreach lib,$(my_static_libraries) $(my_whole_static_libraries), \
       NOTICE-$(if $(LOCAL_IS_HOST_MODULE),HOST,TARGET)-STATIC_LIBRARIES-$(lib))
 
+$(notice_target): | $(installed_static_library_notice_file_targets)
+$(LOCAL_INSTALLED_MODULE): | $(notice_target)
+
 # Default is -fno-rtti.
 ifeq ($(strip $(LOCAL_RTTI_FLAG)),)
 LOCAL_RTTI_FLAG := -fno-rtti
@@ -1820,11 +1823,6 @@ all_libraries := \
     $(built_static_libraries) \
     $(built_whole_libraries)
 
-# Also depend on the notice files for any static libraries that
-# are linked into this module.  This will force them to be installed
-# when this module is.
-$(LOCAL_INSTALLED_MODULE): | $(installed_static_library_notice_file_targets)
-
 ###########################################################
 # Export includes
 ###########################################################
diff --git a/core/build_id.mk b/core/build_id.mk
index 0d9c215a1..5b879c3fd 100644
--- a/core/build_id.mk
+++ b/core/build_id.mk
@@ -18,4 +18,4 @@
 # (like "CRB01").  It must be a single word, and is
 # capitalized by convention.
 
-export BUILD_ID=PPR2.181005.003
+export BUILD_ID=PQ1A.181105.017.A1
diff --git a/core/java_common.mk b/core/java_common.mk
index 2b85dc145..8f4611eed 100644
--- a/core/java_common.mk
+++ b/core/java_common.mk
@@ -422,6 +422,21 @@ $(LOCAL_INTERMEDIATE_TARGETS): PRIVATE_SHARED_JAVA_HEADER_LIBRARIES := $(full_sh
 ALL_MODULES.$(my_register_name).INTERMEDIATE_SOURCE_DIR := \
     $(ALL_MODULES.$(my_register_name).INTERMEDIATE_SOURCE_DIR) $(LOCAL_INTERMEDIATE_SOURCE_DIR)
 
+
+##########################################################
+# Copy NOTICE files of transitive static dependencies
+# Don't do this in mm, since many of the targets won't exist.
+ifeq ($(ONE_SHOT_MAKEFILE),)
+installed_static_library_notice_file_targets := \
+    $(foreach lib,$(LOCAL_STATIC_JAVA_LIBRARIES), \
+      NOTICE-$(if $(LOCAL_IS_HOST_MODULE),HOST,TARGET)-JAVA_LIBRARIES-$(lib))
+else
+installed_static_library_notice_file_targets :=
+endif
+
+$(notice_target): | $(installed_static_library_notice_file_targets)
+$(LOCAL_INSTALLED_MODULE): | $(notice_target)
+
 ###########################################################
 # Verify that all libraries are safe to use
 ###########################################################
diff --git a/core/tasks/sdk-addon.mk b/core/tasks/sdk-addon.mk
index 2a20f070e..bc11b4941 100644
--- a/core/tasks/sdk-addon.mk
+++ b/core/tasks/sdk-addon.mk
@@ -67,6 +67,7 @@ files_to_copy += \
 	$(addon_dir_img):$(INSTALLED_QEMU_VENDORIMAGE):images/$(TARGET_CPU_ABI)/vendor.img \
 	$(addon_dir_img):$(BUILT_RAMDISK_TARGET):images/$(TARGET_CPU_ABI)/ramdisk.img \
 	$(addon_dir_img):$(PRODUCT_OUT)/system/build.prop:images/$(TARGET_CPU_ABI)/build.prop \
+	$(addon_dir_img):device/generic/goldfish/data/etc/userdata.img:images/$(TARGET_CPU_ABI)/userdata.img \
 	$(addon_dir_img):$(target_notice_file_txt):images/$(TARGET_CPU_ABI)/NOTICE.txt \
 	$(addon_dir_img):$(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_SDK_ADDON_SYS_IMG_SOURCE_PROP):images/source.properties
 
diff --git a/core/version_defaults.mk b/core/version_defaults.mk
index 3199c09ca..d2df00e22 100644
--- a/core/version_defaults.mk
+++ b/core/version_defaults.mk
@@ -231,7 +231,7 @@ ifndef PLATFORM_SECURITY_PATCH
     #  It must be of the form "YYYY-MM-DD" on production devices.
     #  It must match one of the Android Security Patch Level strings of the Public Security Bulletins.
     #  If there is no $PLATFORM_SECURITY_PATCH set, keep it empty.
-      PLATFORM_SECURITY_PATCH := 2018-10-05
+      PLATFORM_SECURITY_PATCH := 2018-11-05
 endif
 
 ifndef PLATFORM_SECURITY_PATCH_TIMESTAMP
diff --git a/target/board/generic/device.mk b/target/board/generic/device.mk
index a75bd07ad..0a3241511 100644
--- a/target/board/generic/device.mk
+++ b/target/board/generic/device.mk
@@ -14,17 +14,6 @@
 # limitations under the License.
 #
 
-# This is a build configuration for the product aspects that
-# are specific to the emulator.
-
-PRODUCT_COPY_FILES := \
-    device/generic/goldfish/camera/media_profiles.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_profiles.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_audio.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_codecs_google_audio.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_telephony.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_codecs_google_telephony.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_video.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_codecs_google_video.xml \
-    device/generic/goldfish/camera/media_codecs.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_codecs.xml \
-    hardware/libhardware_legacy/audio/audio_policy.conf:system/etc/audio_policy.conf
-
 # NFC:
 #   Provide default libnfc-nci.conf file for devices that does not have one in
 #   vendor/etc because aosp system image (of aosp_$arch products) is going to
diff --git a/target/board/generic/sepolicy/OWNERS b/target/board/generic/sepolicy/OWNERS
deleted file mode 100644
index 382898894..000000000
--- a/target/board/generic/sepolicy/OWNERS
+++ /dev/null
@@ -1,4 +0,0 @@
-jeffv@google.com
-dcashman@google.com
-jbires@google.com
-sspatil@google.com
diff --git a/target/board/generic/sepolicy/adbd.te b/target/board/generic/sepolicy/adbd.te
deleted file mode 100644
index 9546c1a47..000000000
--- a/target/board/generic/sepolicy/adbd.te
+++ /dev/null
@@ -1 +0,0 @@
-set_prop(adbd, ctl_mdnsd_prop);
diff --git a/target/board/generic/sepolicy/audioserver.te b/target/board/generic/sepolicy/audioserver.te
deleted file mode 100644
index c3c4a3a3d..000000000
--- a/target/board/generic/sepolicy/audioserver.te
+++ /dev/null
@@ -1 +0,0 @@
-allow audioserver bootanim:binder call;
diff --git a/target/board/generic/sepolicy/bootanim.te b/target/board/generic/sepolicy/bootanim.te
deleted file mode 100644
index bc84ee739..000000000
--- a/target/board/generic/sepolicy/bootanim.te
+++ /dev/null
@@ -1,9 +0,0 @@
-allow bootanim self:process execmem;
-allow bootanim ashmem_device:chr_file execute;
-#TODO: This can safely be ignored until b/62954877 is fixed
-dontaudit bootanim system_data_file:dir read;
-
-allow bootanim graphics_device:chr_file { read ioctl open };
-
-typeattribute bootanim system_writes_vendor_properties_violators;
-set_prop(bootanim, qemu_prop)
diff --git a/target/board/generic/sepolicy/cameraserver.te b/target/board/generic/sepolicy/cameraserver.te
deleted file mode 100644
index 6cf5d6ae4..000000000
--- a/target/board/generic/sepolicy/cameraserver.te
+++ /dev/null
@@ -1,2 +0,0 @@
-allow cameraserver system_file:dir { open read };
-allow cameraserver hal_allocator:fd use;
diff --git a/target/board/generic/sepolicy/createns.te b/target/board/generic/sepolicy/createns.te
deleted file mode 100644
index 1eaf9ef58..000000000
--- a/target/board/generic/sepolicy/createns.te
+++ /dev/null
@@ -1,14 +0,0 @@
-# Network namespace creation
-type createns, domain;
-type createns_exec, exec_type, vendor_file_type, file_type;
-
-init_daemon_domain(createns)
-
-allow createns self:capability { sys_admin net_raw setuid setgid };
-allow createns varrun_file:dir { add_name search write };
-allow createns varrun_file:file { create mounton open read write };
-
-#Allow createns itself to be run by init in its own domain
-domain_auto_trans(goldfish_setup, createns_exec, createns);
-allow createns goldfish_setup:fd use;
-
diff --git a/target/board/generic/sepolicy/device.te b/target/board/generic/sepolicy/device.te
deleted file mode 100644
index d12944119..000000000
--- a/target/board/generic/sepolicy/device.te
+++ /dev/null
@@ -1 +0,0 @@
-type qemu_device, dev_type, mlstrustedobject;
diff --git a/target/board/generic/sepolicy/dhcpclient.te b/target/board/generic/sepolicy/dhcpclient.te
deleted file mode 100644
index df71fca38..000000000
--- a/target/board/generic/sepolicy/dhcpclient.te
+++ /dev/null
@@ -1,20 +0,0 @@
-# DHCP client
-type dhcpclient, domain;
-type dhcpclient_exec, exec_type, vendor_file_type, file_type;
-
-init_daemon_domain(dhcpclient)
-net_domain(dhcpclient)
-
-allow dhcpclient execns:fd use;
-
-set_prop(dhcpclient, net_eth0_prop);
-allow dhcpclient self:capability { net_admin net_raw };
-allow dhcpclient self:udp_socket create;
-allow dhcpclient self:netlink_route_socket { write nlmsg_write };
-allow dhcpclient varrun_file:dir search;
-allow dhcpclient self:packet_socket { create bind write read };
-allowxperm dhcpclient self:udp_socket ioctl { SIOCSIFFLAGS
-                                              SIOCSIFADDR
-                                              SIOCSIFNETMASK
-                                              SIOCSIFMTU
-                                              SIOCGIFHWADDR };
diff --git a/target/board/generic/sepolicy/dhcpserver.te b/target/board/generic/sepolicy/dhcpserver.te
deleted file mode 100644
index 7e8ba263a..000000000
--- a/target/board/generic/sepolicy/dhcpserver.te
+++ /dev/null
@@ -1,12 +0,0 @@
-# DHCP server
-type dhcpserver, domain;
-type dhcpserver_exec, exec_type, vendor_file_type, file_type;
-
-init_daemon_domain(dhcpserver)
-net_domain(dhcpserver)
-
-allow dhcpserver execns:fd use;
-
-get_prop(dhcpserver, net_eth0_prop);
-allow dhcpserver self:udp_socket { ioctl create setopt bind };
-allow dhcpserver self:capability { net_raw net_bind_service };
diff --git a/target/board/generic/sepolicy/domain.te b/target/board/generic/sepolicy/domain.te
deleted file mode 100644
index 3706dbaa0..000000000
--- a/target/board/generic/sepolicy/domain.te
+++ /dev/null
@@ -1,3 +0,0 @@
-allow domain qemu_device:chr_file rw_file_perms;
-
-get_prop(domain, qemu_prop)
diff --git a/target/board/generic/sepolicy/execns.te b/target/board/generic/sepolicy/execns.te
deleted file mode 100644
index 9675a99c7..000000000
--- a/target/board/generic/sepolicy/execns.te
+++ /dev/null
@@ -1,34 +0,0 @@
-# Network namespace transitions
-type execns, domain;
-type execns_exec, exec_type, vendor_file_type, file_type;
-
-init_daemon_domain(execns)
-
-allow execns varrun_file:dir search;
-allow execns varrun_file:file r_file_perms;
-allow execns self:capability sys_admin;
-allow execns nsfs:file { open read };
-
-#Allow execns itself to be run by init in its own domain
-domain_auto_trans(init, execns_exec, execns);
-
-# Allow dhcpclient to be run by execns in its own domain
-domain_auto_trans(execns, dhcpclient_exec, dhcpclient);
-
-# Allow dhcpserver to be run by execns in its own domain
-domain_auto_trans(execns, dhcpserver_exec, dhcpserver);
-
-# Rules to allow execution of hostapd and allow it to run
-allow execns hal_wifi_hostapd_default_exec:file { execute_no_trans };
-allow execns self:capability { net_admin net_raw };
-allow execns self:netlink_generic_socket { bind create getattr read setopt write };
-allow execns self:netlink_route_socket { bind create read write nlmsg_write };
-allow execns execns:udp_socket { create ioctl };
-allow execns self:packet_socket { create setopt };
-allow execns sysfs_net:dir { search };
-allowxperm execns self:udp_socket ioctl priv_sock_ioctls;
-
-# Allow execns to read createns proc file to get the namespace file
-allow execns createns:file read;
-allow execns createns:dir search;
-allow execns createns:lnk_file read;
diff --git a/target/board/generic/sepolicy/file.te b/target/board/generic/sepolicy/file.te
deleted file mode 100644
index b0aa217ae..000000000
--- a/target/board/generic/sepolicy/file.te
+++ /dev/null
@@ -1,4 +0,0 @@
-type sysfs_writable, fs_type, sysfs_type, mlstrustedobject;
-type varrun_file, file_type, data_file_type, mlstrustedobject;
-type mediadrm_vendor_data_file, file_type, data_file_type;
-type nsfs, fs_type;
diff --git a/target/board/generic/sepolicy/file_contexts b/target/board/generic/sepolicy/file_contexts
deleted file mode 100644
index 73fe75245..000000000
--- a/target/board/generic/sepolicy/file_contexts
+++ /dev/null
@@ -1,46 +0,0 @@
-# goldfish
-/dev/block/mtdblock0         u:object_r:system_block_device:s0
-/dev/block/mtdblock1         u:object_r:userdata_block_device:s0
-/dev/block/mtdblock2         u:object_r:cache_block_device:s0
-
-# ranchu
-/dev/block/vda               u:object_r:system_block_device:s0
-/dev/block/vdb               u:object_r:cache_block_device:s0
-/dev/block/vdc               u:object_r:userdata_block_device:s0
-/dev/block/vdd               u:object_r:metadata_block_device:s0
-/dev/block/vde               u:object_r:system_block_device:s0
-
-/dev/goldfish_pipe           u:object_r:qemu_device:s0
-/dev/goldfish_sync           u:object_r:qemu_device:s0
-/dev/qemu_.*                 u:object_r:qemu_device:s0
-/dev/ttyGF[0-9]*             u:object_r:serial_device:s0
-/dev/ttyS2                   u:object_r:console_device:s0
-/vendor/bin/init\.ranchu-core\.sh u:object_r:goldfish_setup_exec:s0
-/vendor/bin/init\.ranchu-net\.sh u:object_r:goldfish_setup_exec:s0
-/vendor/bin/init\.wifi\.sh   u:object_r:goldfish_setup_exec:s0
-/vendor/bin/qemu-props       u:object_r:qemu_props_exec:s0
-/vendor/bin/createns         u:object_r:createns_exec:s0
-/vendor/bin/execns           u:object_r:execns_exec:s0
-/vendor/bin/ipv6proxy        u:object_r:ipv6proxy_exec:s0
-/vendor/bin/dhcpclient       u:object_r:dhcpclient_exec:s0
-/vendor/bin/dhcpserver       u:object_r:dhcpserver_exec:s0
-
-/vendor/bin/hw/android\.hardware\.drm@1\.0-service\.widevine          u:object_r:hal_drm_widevine_exec:s0
-
-/vendor/lib(64)?/hw/gralloc\.ranchu\.so   u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/hw/gralloc\.goldfish\.default\.so   u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libEGL_emulation\.so          u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libGLESv1_CM_emulation\.so    u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libGLESv2_emulation\.so       u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libEGL_swiftshader\.so          u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libGLESv1_CM_swiftshader\.so    u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libGLESv2_swiftshader\.so       u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libOpenglSystemCommon\.so       u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/lib_renderControl_enc\.so       u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libGLESv1_enc\.so       u:object_r:same_process_hal_file:s0
-/vendor/lib(64)?/libGLESv2_enc\.so       u:object_r:same_process_hal_file:s0
-
-# data
-/data/vendor/mediadrm(/.*)?            u:object_r:mediadrm_vendor_data_file:s0
-/data/vendor/var/run(/.*)?             u:object_r:varrun_file:s0
-
diff --git a/target/board/generic/sepolicy/genfs_contexts b/target/board/generic/sepolicy/genfs_contexts
deleted file mode 100644
index 1b816263b..000000000
--- a/target/board/generic/sepolicy/genfs_contexts
+++ /dev/null
@@ -1,20 +0,0 @@
-# On the emulator, device tree dir is configured to be
-# /sys/bus/platform/devices/ANDR0001:00/properties/android/ which is a symlink to
-# /sys/devices/platform/ANDR0001:00/properties/android/
-genfscon sysfs /devices/platform/ANDR0001:00/properties/android u:object_r:sysfs_dt_firmware_android:s0
-
-# We expect /sys/class/power_supply/* and everything it links to to be labeled
-# as sysfs_batteryinfo.
-genfscon sysfs /devices/platform/GFSH0001:00/power_supply u:object_r:sysfs_batteryinfo:s0
-
-# /sys/class/rtc
-genfscon sysfs /devices/pnp0/00:00/rtc u:object_r:sysfs_rtc:s0
-genfscon sysfs /devices/platform/GFSH0007:00/rtc u:object_r:sysfs_rtc:s0
-
-# /sys/class/net
-genfscon sysfs /devices/pci0000:00/0000:00:08.0/virtio5/net u:object_r:sysfs_net:s0
-genfscon sysfs /devices/virtual/mac80211_hwsim/hwsim0/net u:object_r:sysfs_net:s0
-genfscon sysfs /devices/virtual/mac80211_hwsim/hwsim1/net u:object_r:sysfs_net:s0
-
-# /proc/<pid>/ns
-genfscon nsfs / u:object_r:nsfs:s0
diff --git a/target/board/generic/sepolicy/goldfish_setup.te b/target/board/generic/sepolicy/goldfish_setup.te
deleted file mode 100644
index 3041436b3..000000000
--- a/target/board/generic/sepolicy/goldfish_setup.te
+++ /dev/null
@@ -1,47 +0,0 @@
-# goldfish-setup service: runs init.goldfish.sh script
-type goldfish_setup, domain;
-type goldfish_setup_exec, vendor_file_type, exec_type, file_type;
-
-init_daemon_domain(goldfish_setup)
-
-# TODO(b/79502552): Invalid property access from emulator vendor
-#set_prop(goldfish_setup, debug_prop);
-allow goldfish_setup self:capability { net_admin net_raw };
-allow goldfish_setup self:udp_socket { create ioctl };
-allow goldfish_setup vendor_toolbox_exec:file execute_no_trans;
-allowxperm goldfish_setup self:udp_socket ioctl priv_sock_ioctls;
-wakelock_use(goldfish_setup);
-allow goldfish_setup vendor_shell_exec:file { rx_file_perms };
-
-# Set system properties to start services
-set_prop(goldfish_setup, ctl_default_prop);
-
-# Set up WiFi
-allow goldfish_setup self:netlink_route_socket { create nlmsg_write setopt bind getattr read write nlmsg_read };
-allow goldfish_setup self:netlink_generic_socket create_socket_perms_no_ioctl;
-allow goldfish_setup self:capability { sys_module sys_admin };
-allow goldfish_setup varrun_file:dir { mounton open read write add_name search remove_name };
-allow goldfish_setup varrun_file:file { mounton getattr create read write open unlink };
-allow goldfish_setup execns_exec:file rx_file_perms;
-allow goldfish_setup proc_net:file rw_file_perms;
-allow goldfish_setup proc:file r_file_perms;
-allow goldfish_setup nsfs:file r_file_perms;
-allow goldfish_setup system_data_file:dir getattr;
-allow goldfish_setup kernel:system module_request;
-set_prop(goldfish_setup, qemu_prop);
-get_prop(goldfish_setup, net_share_prop);
-# Allow goldfish_setup to run /system/bin/ip and /system/bin/iw
-allow goldfish_setup system_file:file execute_no_trans;
-# Allow goldfish_setup to run init.wifi.sh
-allow goldfish_setup goldfish_setup_exec:file execute_no_trans;
-#Allow goldfish_setup to run createns in its own domain
-domain_auto_trans(goldfish_setup, createns_exec, createns);
-# iw
-allow goldfish_setup sysfs:file { read open };
-# iptables
-allow goldfish_setup system_file:file lock;
-allow goldfish_setup self:rawip_socket { create getopt setopt };
-# Allow goldfish_setup to read createns proc file to get the namespace file
-allow goldfish_setup createns:file { read };
-allow goldfish_setup createns:dir { search };
-allow goldfish_setup createns:lnk_file { read };
diff --git a/target/board/generic/sepolicy/hal_camera_default.te b/target/board/generic/sepolicy/hal_camera_default.te
deleted file mode 100644
index eb88c36f0..000000000
--- a/target/board/generic/sepolicy/hal_camera_default.te
+++ /dev/null
@@ -1,3 +0,0 @@
-vndbinder_use(hal_camera_default);
-allow hal_camera_default hal_graphics_mapper_hwservice:hwservice_manager find;
-hal_client_domain(hal_camera_default, hal_graphics_composer)
diff --git a/target/board/generic/sepolicy/hal_cas_default.te b/target/board/generic/sepolicy/hal_cas_default.te
deleted file mode 100644
index 3ed3bee86..000000000
--- a/target/board/generic/sepolicy/hal_cas_default.te
+++ /dev/null
@@ -1 +0,0 @@
-vndbinder_use(hal_cas_default);
diff --git a/target/board/generic/sepolicy/hal_drm_default.te b/target/board/generic/sepolicy/hal_drm_default.te
deleted file mode 100644
index 5a07433c8..000000000
--- a/target/board/generic/sepolicy/hal_drm_default.te
+++ /dev/null
@@ -1,2 +0,0 @@
-vndbinder_use(hal_drm_default);
-hal_client_domain(hal_drm_default, hal_graphics_composer)
diff --git a/target/board/generic/sepolicy/hal_drm_widevine.te b/target/board/generic/sepolicy/hal_drm_widevine.te
deleted file mode 100644
index d49000d14..000000000
--- a/target/board/generic/sepolicy/hal_drm_widevine.te
+++ /dev/null
@@ -1,14 +0,0 @@
-# define SELinux domain
-type hal_drm_widevine, domain;
-hal_server_domain(hal_drm_widevine, hal_drm)
-
-type hal_drm_widevine_exec, exec_type, vendor_file_type, file_type;
-init_daemon_domain(hal_drm_widevine)
-
-allow hal_drm mediacodec:fd use;
-allow hal_drm { appdomain -isolated_app }:fd use;
-
-vndbinder_use(hal_drm_widevine);
-hal_client_domain(hal_drm_widevine, hal_graphics_composer);
-allow hal_drm_widevine mediadrm_vendor_data_file:dir create_dir_perms;
-allow hal_drm_widevine mediadrm_vendor_data_file:file create_file_perms;
diff --git a/target/board/generic/sepolicy/hal_fingerprint_default.te b/target/board/generic/sepolicy/hal_fingerprint_default.te
deleted file mode 100644
index e5b06f12d..000000000
--- a/target/board/generic/sepolicy/hal_fingerprint_default.te
+++ /dev/null
@@ -1,5 +0,0 @@
-# TODO(b/36644492): Remove data_between_core_and_vendor_violators once
-# hal_fingerprint no longer directly accesses fingerprintd_data_file.
-typeattribute hal_fingerprint_default data_between_core_and_vendor_violators;
-allow hal_fingerprint_default fingerprintd_data_file:file create_file_perms;
-allow hal_fingerprint_default fingerprintd_data_file:dir rw_dir_perms;
diff --git a/target/board/generic/sepolicy/hal_gnss_default.te b/target/board/generic/sepolicy/hal_gnss_default.te
deleted file mode 100644
index 0dd3d0356..000000000
--- a/target/board/generic/sepolicy/hal_gnss_default.te
+++ /dev/null
@@ -1,3 +0,0 @@
-#============= hal_gnss_default ==============
-allow hal_gnss_default vndbinder_device:chr_file { ioctl open read write };
-
diff --git a/target/board/generic/sepolicy/hal_graphics_allocator_default.te b/target/board/generic/sepolicy/hal_graphics_allocator_default.te
deleted file mode 100644
index 0c8e27de9..000000000
--- a/target/board/generic/sepolicy/hal_graphics_allocator_default.te
+++ /dev/null
@@ -1,2 +0,0 @@
-allow hal_graphics_allocator_default graphics_device:dir search;
-allow hal_graphics_allocator_default graphics_device:chr_file { ioctl open read write };
diff --git a/target/board/generic/sepolicy/hal_graphics_composer_default.te b/target/board/generic/sepolicy/hal_graphics_composer_default.te
deleted file mode 100644
index 034bdeff9..000000000
--- a/target/board/generic/sepolicy/hal_graphics_composer_default.te
+++ /dev/null
@@ -1,3 +0,0 @@
-#============= hal_graphics_composer_default ==============
-allow hal_graphics_composer_default vndbinder_device:chr_file { ioctl open read write };
-
diff --git a/target/board/generic/sepolicy/hal_wifi_default.te b/target/board/generic/sepolicy/hal_wifi_default.te
deleted file mode 100644
index de4b9969b..000000000
--- a/target/board/generic/sepolicy/hal_wifi_default.te
+++ /dev/null
@@ -1 +0,0 @@
-allow hal_wifi_default hal_wifi_default:netlink_route_socket { create bind write read nlmsg_read };
diff --git a/target/board/generic/sepolicy/healthd.te b/target/board/generic/sepolicy/healthd.te
deleted file mode 100644
index ced670499..000000000
--- a/target/board/generic/sepolicy/healthd.te
+++ /dev/null
@@ -1,2 +0,0 @@
-# Allow to read /sys/class/power_supply directory
-allow healthd sysfs:dir r_dir_perms;
diff --git a/target/board/generic/sepolicy/init.te b/target/board/generic/sepolicy/init.te
deleted file mode 100644
index 84a4e8dbf..000000000
--- a/target/board/generic/sepolicy/init.te
+++ /dev/null
@@ -1,2 +0,0 @@
-allow init tmpfs:lnk_file create_file_perms;
-dontaudit init kernel:system module_request;
diff --git a/target/board/generic/sepolicy/ipv6proxy.te b/target/board/generic/sepolicy/ipv6proxy.te
deleted file mode 100644
index 22976fe9b..000000000
--- a/target/board/generic/sepolicy/ipv6proxy.te
+++ /dev/null
@@ -1,16 +0,0 @@
-# IPv6 proxying
-type ipv6proxy, domain;
-type ipv6proxy_exec, exec_type, vendor_file_type, file_type;
-
-init_daemon_domain(ipv6proxy)
-net_domain(ipv6proxy)
-
-# Allow ipv6proxy to be run by execns in its own domain
-domain_auto_trans(execns, ipv6proxy_exec, ipv6proxy);
-allow ipv6proxy execns:fd use;
-
-allow ipv6proxy self:capability { sys_admin sys_module net_admin net_raw };
-allow ipv6proxy self:packet_socket { bind create read };
-allow ipv6proxy self:netlink_route_socket nlmsg_write;
-allow ipv6proxy varrun_file:dir search;
-allowxperm ipv6proxy self:udp_socket ioctl { SIOCSIFFLAGS SIOCGIFHWADDR };
diff --git a/target/board/generic/sepolicy/logpersist.te b/target/board/generic/sepolicy/logpersist.te
deleted file mode 100644
index 3fc025019..000000000
--- a/target/board/generic/sepolicy/logpersist.te
+++ /dev/null
@@ -1,13 +0,0 @@
-# goldfish logcat service:  runs logcat -Q in logpersist domain
-
-# See global logcat.te/logpersist.te, only set for eng & userdebug,
-# allow for all builds in a non-conflicting manner.
-
-domain_auto_trans(init, logcat_exec, logpersist)
-
-# Read from logd.
-unix_socket_connect(logpersist, logdr, logd)
-
-# Write to /dev/ttyS2 and /dev/ttyGF2.
-allow logpersist serial_device:chr_file { write open };
-get_prop(logpersist, qemu_cmdline)
diff --git a/target/board/generic/sepolicy/mediacodec.te b/target/board/generic/sepolicy/mediacodec.te
deleted file mode 100644
index acf4e59b9..000000000
--- a/target/board/generic/sepolicy/mediacodec.te
+++ /dev/null
@@ -1 +0,0 @@
-allow mediacodec system_file:dir { open read };
diff --git a/target/board/generic/sepolicy/netd.te b/target/board/generic/sepolicy/netd.te
deleted file mode 100644
index 09a28b996..000000000
--- a/target/board/generic/sepolicy/netd.te
+++ /dev/null
@@ -1,3 +0,0 @@
-dontaudit netd self:capability sys_module;
-#TODO: This can safely be ignored until b/62954877 is fixed
-dontaudit netd kernel:system module_request;
diff --git a/target/board/generic/sepolicy/priv_app.te b/target/board/generic/sepolicy/priv_app.te
deleted file mode 100644
index 3d16f32b0..000000000
--- a/target/board/generic/sepolicy/priv_app.te
+++ /dev/null
@@ -1,5 +0,0 @@
-#TODO: b/62908025
-dontaudit priv_app firstboot_prop:file { getattr open };
-dontaudit priv_app device:dir { open read };
-dontaudit priv_app proc_interrupts:file { getattr open read };
-dontaudit priv_app proc_modules:file { getattr open read };
diff --git a/target/board/generic/sepolicy/property.te b/target/board/generic/sepolicy/property.te
deleted file mode 100644
index 3593a39dd..000000000
--- a/target/board/generic/sepolicy/property.te
+++ /dev/null
@@ -1,5 +0,0 @@
-type qemu_prop, property_type;
-type qemu_cmdline, property_type;
-type radio_noril_prop, property_type;
-type net_eth0_prop, property_type;
-type net_share_prop, property_type;
diff --git a/target/board/generic/sepolicy/property_contexts b/target/board/generic/sepolicy/property_contexts
deleted file mode 100644
index f7a241cfb..000000000
--- a/target/board/generic/sepolicy/property_contexts
+++ /dev/null
@@ -1,8 +0,0 @@
-qemu.                   u:object_r:qemu_prop:s0
-qemu.cmdline            u:object_r:qemu_cmdline:s0
-vendor.qemu		u:object_r:qemu_prop:s0
-ro.emu.                 u:object_r:qemu_prop:s0
-ro.emulator.            u:object_r:qemu_prop:s0
-ro.radio.noril          u:object_r:radio_noril_prop:s0
-net.eth0.               u:object_r:net_eth0_prop:s0
-net.shared_net_ip       u:object_r:net_share_prop:s0
diff --git a/target/board/generic/sepolicy/qemu_props.te b/target/board/generic/sepolicy/qemu_props.te
deleted file mode 100644
index b3e2d9552..000000000
--- a/target/board/generic/sepolicy/qemu_props.te
+++ /dev/null
@@ -1,10 +0,0 @@
-# qemu-props service:  Sets system properties on boot.
-type qemu_props, domain;
-type qemu_props_exec, vendor_file_type, exec_type, file_type;
-
-init_daemon_domain(qemu_props)
-
-set_prop(qemu_props, qemu_prop)
-# TODO(b/79502552): Invalid property access from emulator vendor
-#set_prop(qemu_props, qemu_cmdline)
-set_prop(qemu_props, qemu_cmdline)
diff --git a/target/board/generic/sepolicy/radio.te b/target/board/generic/sepolicy/radio.te
deleted file mode 100644
index 742d3b2d2..000000000
--- a/target/board/generic/sepolicy/radio.te
+++ /dev/null
@@ -1,3 +0,0 @@
-# Allow the radio to read these properties, they only have an SELinux label in
-# the emulator.
-get_prop(radio, net_eth0_prop);
diff --git a/target/board/generic/sepolicy/rild.te b/target/board/generic/sepolicy/rild.te
deleted file mode 100644
index ea183739a..000000000
--- a/target/board/generic/sepolicy/rild.te
+++ /dev/null
@@ -1,3 +0,0 @@
-# Allow rild to read these properties, they only have an SELinux label in the
-# emulator.
-get_prop(rild, net_eth0_prop);
diff --git a/target/board/generic/sepolicy/shell.te b/target/board/generic/sepolicy/shell.te
deleted file mode 100644
index b246d7e3c..000000000
--- a/target/board/generic/sepolicy/shell.te
+++ /dev/null
@@ -1 +0,0 @@
-allow shell serial_device:chr_file rw_file_perms;
diff --git a/target/board/generic/sepolicy/surfaceflinger.te b/target/board/generic/sepolicy/surfaceflinger.te
deleted file mode 100644
index 2bba8a78b..000000000
--- a/target/board/generic/sepolicy/surfaceflinger.te
+++ /dev/null
@@ -1,5 +0,0 @@
-allow surfaceflinger self:process execmem;
-allow surfaceflinger ashmem_device:chr_file execute;
-
-typeattribute surfaceflinger system_writes_vendor_properties_violators;
-set_prop(surfaceflinger, qemu_prop)
diff --git a/target/board/generic/sepolicy/system_server.te b/target/board/generic/sepolicy/system_server.te
deleted file mode 100644
index dd70b12dd..000000000
--- a/target/board/generic/sepolicy/system_server.te
+++ /dev/null
@@ -1 +0,0 @@
-get_prop(system_server, radio_noril_prop)
diff --git a/target/board/generic/sepolicy/vendor_init.te b/target/board/generic/sepolicy/vendor_init.te
deleted file mode 100644
index b18d3913f..000000000
--- a/target/board/generic/sepolicy/vendor_init.te
+++ /dev/null
@@ -1 +0,0 @@
-set_prop(vendor_init, qemu_prop)
diff --git a/target/board/generic/sepolicy/vold.te b/target/board/generic/sepolicy/vold.te
deleted file mode 100644
index 5f3bdd446..000000000
--- a/target/board/generic/sepolicy/vold.te
+++ /dev/null
@@ -1 +0,0 @@
-dontaudit vold kernel:system module_request;
diff --git a/target/board/generic/sepolicy/zygote.te b/target/board/generic/sepolicy/zygote.te
deleted file mode 100644
index da403b5dd..000000000
--- a/target/board/generic/sepolicy/zygote.te
+++ /dev/null
@@ -1,5 +0,0 @@
-typeattribute zygote system_writes_vendor_properties_violators;
-set_prop(zygote, qemu_prop)
-# TODO (b/63631799) fix this access
-# Suppress denials to storage. Webview zygote should not be accessing.
-dontaudit webview_zygote mnt_expand_file:dir getattr;
diff --git a/target/board/generic_arm64/device.mk b/target/board/generic_arm64/device.mk
index 8bd6a8bc5..2004624ca 100644
--- a/target/board/generic_arm64/device.mk
+++ b/target/board/generic_arm64/device.mk
@@ -14,16 +14,6 @@
 # limitations under the License.
 #
 
-# This is a build configuration for the product aspects that
-# are specific to the emulator.
-
-PRODUCT_COPY_FILES := \
-    device/generic/goldfish/camera/media_profiles.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_profiles.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_audio.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_codecs_google_audio.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_telephony.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_codecs_google_telephony.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_video.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_codecs_google_video.xml \
-    device/generic/goldfish/camera/media_codecs.xml:$(TARGET_COPY_OUT_VENDOR)/etc/media_codecs.xml
-
 # NFC:
 #   Provide default libnfc-nci.conf file for devices that does not have one in
 #   vendor/etc because aosp system image (of aosp_$arch products) is going to
diff --git a/target/board/generic_x86/device.mk b/target/board/generic_x86/device.mk
index fa2d4729a..0a3241511 100644
--- a/target/board/generic_x86/device.mk
+++ b/target/board/generic_x86/device.mk
@@ -14,16 +14,6 @@
 # limitations under the License.
 #
 
-# This is a build configuration for the product aspects that
-# are specific to the emulator.
-
-PRODUCT_COPY_FILES := \
-    device/generic/goldfish/camera/media_profiles.xml:system/etc/media_profiles.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_audio.xml:system/etc/media_codecs_google_audio.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_telephony.xml:system/etc/media_codecs_google_telephony.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_video.xml:system/etc/media_codecs_google_video.xml \
-    device/generic/goldfish/camera/media_codecs.xml:system/etc/media_codecs.xml
-
 # NFC:
 #   Provide default libnfc-nci.conf file for devices that does not have one in
 #   vendor/etc because aosp system image (of aosp_$arch products) is going to
@@ -32,7 +22,3 @@ PRODUCT_COPY_FILES := \
 #   NFC configuration file should be in vendor/etc, instead of system/etc
 PRODUCT_COPY_FILES += \
     device/generic/common/nfc/libnfc-nci.conf:system/etc/libnfc-nci.conf
-
-PRODUCT_PACKAGES := \
-    audio.primary.goldfish \
-    vibrator.goldfish
diff --git a/target/board/generic_x86/sepolicy/OWNERS b/target/board/generic_x86/sepolicy/OWNERS
deleted file mode 100644
index 382898894..000000000
--- a/target/board/generic_x86/sepolicy/OWNERS
+++ /dev/null
@@ -1,4 +0,0 @@
-jeffv@google.com
-dcashman@google.com
-jbires@google.com
-sspatil@google.com
diff --git a/target/board/generic_x86/sepolicy/domain.te b/target/board/generic_x86/sepolicy/domain.te
deleted file mode 100644
index 0bc8d871f..000000000
--- a/target/board/generic_x86/sepolicy/domain.te
+++ /dev/null
@@ -1 +0,0 @@
-allow domain cpuctl_device:dir search;
diff --git a/target/board/generic_x86/sepolicy/healthd.te b/target/board/generic_x86/sepolicy/healthd.te
deleted file mode 100644
index 95fa8079b..000000000
--- a/target/board/generic_x86/sepolicy/healthd.te
+++ /dev/null
@@ -1 +0,0 @@
-allow healthd self:capability sys_nice;
diff --git a/target/board/generic_x86/sepolicy/init.te b/target/board/generic_x86/sepolicy/init.te
deleted file mode 100644
index 3aa81d1b5..000000000
--- a/target/board/generic_x86/sepolicy/init.te
+++ /dev/null
@@ -1 +0,0 @@
-allow init tmpfs:lnk_file create_file_perms;
diff --git a/target/board/generic_x86/sepolicy/installd.te b/target/board/generic_x86/sepolicy/installd.te
deleted file mode 100644
index 7a558b129..000000000
--- a/target/board/generic_x86/sepolicy/installd.te
+++ /dev/null
@@ -1 +0,0 @@
-allow installd self:process execmem;
diff --git a/target/board/generic_x86/sepolicy/zygote.te b/target/board/generic_x86/sepolicy/zygote.te
deleted file mode 100644
index 93993a47f..000000000
--- a/target/board/generic_x86/sepolicy/zygote.te
+++ /dev/null
@@ -1,2 +0,0 @@
-allow zygote self:process execmem;
-allow zygote self:capability sys_nice;
diff --git a/target/board/generic_x86_64/device.mk b/target/board/generic_x86_64/device.mk
index fa2d4729a..0a3241511 100755
--- a/target/board/generic_x86_64/device.mk
+++ b/target/board/generic_x86_64/device.mk
@@ -14,16 +14,6 @@
 # limitations under the License.
 #
 
-# This is a build configuration for the product aspects that
-# are specific to the emulator.
-
-PRODUCT_COPY_FILES := \
-    device/generic/goldfish/camera/media_profiles.xml:system/etc/media_profiles.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_audio.xml:system/etc/media_codecs_google_audio.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_telephony.xml:system/etc/media_codecs_google_telephony.xml \
-    frameworks/av/media/libstagefright/data/media_codecs_google_video.xml:system/etc/media_codecs_google_video.xml \
-    device/generic/goldfish/camera/media_codecs.xml:system/etc/media_codecs.xml
-
 # NFC:
 #   Provide default libnfc-nci.conf file for devices that does not have one in
 #   vendor/etc because aosp system image (of aosp_$arch products) is going to
@@ -32,7 +22,3 @@ PRODUCT_COPY_FILES := \
 #   NFC configuration file should be in vendor/etc, instead of system/etc
 PRODUCT_COPY_FILES += \
     device/generic/common/nfc/libnfc-nci.conf:system/etc/libnfc-nci.conf
-
-PRODUCT_PACKAGES := \
-    audio.primary.goldfish \
-    vibrator.goldfish
diff --git a/target/board/generic_x86_arm/BoardConfig.mk b/target/board/generic_x86_arm/BoardConfig.mk
index 131c0014f..c66aacc84 100644
--- a/target/board/generic_x86_arm/BoardConfig.mk
+++ b/target/board/generic_x86_arm/BoardConfig.mk
@@ -61,4 +61,4 @@ BOARD_CACHEIMAGE_FILE_SYSTEM_TYPE := ext4
 BOARD_FLASH_BLOCK_SIZE := 512
 TARGET_USERIMAGES_SPARSE_EXT_DISABLED := true
 
-BOARD_SEPOLICY_DIRS += build/target/board/generic/sepolicy
+BOARD_SEPOLICY_DIRS += device/generic/goldfish/sepolicy/common
diff --git a/target/product/product_launched_with_k.mk b/target/product/product_launched_with_k.mk
new file mode 100644
index 000000000..87faa12a6
--- /dev/null
+++ b/target/product/product_launched_with_k.mk
@@ -0,0 +1,2 @@
+#PRODUCT_SHIPPING_API_LEVEL indicates the first api level, device has been commercially launched on.
+PRODUCT_SHIPPING_API_LEVEL := 19
diff --git a/target/product/product_launched_with_l.mk b/target/product/product_launched_with_l.mk
index 6e782f798..4e79749a3 100644
--- a/target/product/product_launched_with_l.mk
+++ b/target/product/product_launched_with_l.mk
@@ -1,3 +1,2 @@
 #PRODUCT_SHIPPING_API_LEVEL indicates the first api level, device has been commercially launched on.
 PRODUCT_SHIPPING_API_LEVEL := 21
-
diff --git a/target/product/sdk_phone_arm64.mk b/target/product/sdk_phone_arm64.mk
index 56eb8c791..c6b290f79 100644
--- a/target/product/sdk_phone_arm64.mk
+++ b/target/product/sdk_phone_arm64.mk
@@ -14,31 +14,11 @@
 # limitations under the License.
 #
 
-PRODUCT_PROPERTY_OVERRIDES += \
-	rild.libpath=/vendor/lib64/libreference-ril.so
+$(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_arm64.mk)
 
-# This is a build configuration for a full-featured build of the
-# Open-Source part of the tree. It's geared toward a US-centric
-# build quite specifically for the emulator, and might not be
-# entirely appropriate to inherit from for on-device configurations.
-
-# Note: the following lines need to stay at the beginning so that it can
-# take priority  and override the rules it inherit from other mk files
-# see copy file rules in core/Makefile
-PRODUCT_COPY_FILES += \
-    development/sys-img/advancedFeatures.ini.arm:advancedFeatures.ini \
-    prebuilts/qemu-kernel/arm64/3.18/kernel-qemu2:kernel-ranchu \
-    device/generic/goldfish/fstab.ranchu.arm:$(TARGET_COPY_OUT_VENDOR)/etc/fstab.ranchu
-
-$(call inherit-product, $(SRC_TARGET_DIR)/product/core_64_bit.mk)
-$(call inherit-product, $(SRC_TARGET_DIR)/product/sdk_base.mk)
-$(call inherit-product, $(SRC_TARGET_DIR)/board/generic_arm64/device.mk)
-
-# AOSP emulator images build the AOSP messaging app.
-# Google API images override with the Google API app.
-# See vendor/google/products/sdk_google_phone_*.mk
-PRODUCT_PACKAGES += \
-    messaging
+# Define the host tools and libs that are parts of the SDK.
+$(call inherit-product, sdk/build/product_sdk.mk)
+$(call inherit-product, development/build/product_sdk.mk)
 
 # Overrides
 PRODUCT_BRAND := Android
diff --git a/target/product/sdk_phone_armv7.mk b/target/product/sdk_phone_armv7.mk
index 73c42c381..f2b51cfd7 100644
--- a/target/product/sdk_phone_armv7.mk
+++ b/target/product/sdk_phone_armv7.mk
@@ -14,24 +14,12 @@
 # limitations under the License.
 #
 
-PRODUCT_PROPERTY_OVERRIDES += \
-	rild.libpath=/vendor/lib/libreference-ril.so
+$(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_arm.mk)
 
-# Note: the following lines need to stay at the beginning so that it can
-# take priority  and override the rules it inherit from other mk files
-# see copy file rules in core/Makefile
-PRODUCT_COPY_FILES += \
-    development/sys-img/advancedFeatures.ini.arm:advancedFeatures.ini \
-    prebuilts/qemu-kernel/arm64/3.18/kernel-qemu2:kernel-ranchu-64 \
-    device/generic/goldfish/fstab.ranchu.arm:$(TARGET_COPY_OUT_VENDOR)/etc/fstab.ranchu
+# Define the host tools and libs that are parts of the SDK.
+$(call inherit-product, sdk/build/product_sdk.mk)
+$(call inherit-product, development/build/product_sdk.mk)
 
-$(call inherit-product, $(SRC_TARGET_DIR)/product/sdk_base.mk)
-
-# AOSP emulator images build the AOSP messaging app.
-# Google API images override with the Google API app.
-# See vendor/google/products/sdk_google_phone_*.mk
-PRODUCT_PACKAGES += \
-    messaging
 
 # Overrides
 PRODUCT_BRAND := Android
-- 
2.17.1

